src/mongocrypt-ctx-encrypt.c

```diff
@@ -357,6 +357,43 @@ _fle2_collect_keys_for_deleteTokens (mongocrypt_ctx_t *ctx)
    return true;
 }
 
+/* _fle2_collect_keys_for_compact requests keys required to produce
+ * compactionTokens. compactionTokens is only applicable to FLE 2. */
+static bool
+_fle2_collect_keys_for_compact (mongocrypt_ctx_t *ctx)
+{
+   _mongocrypt_ctx_encrypt_t *ectx = (_mongocrypt_ctx_encrypt_t *) ctx;
+
+   /* compactionTokens are only appended for FLE 2. */
+   if (_mongocrypt_buffer_empty (&ectx->encrypted_field_config)) {
+      return true;
+   }
```

Comment:
I've seen this "check for fle2" a few times now spelled as checking whether `encrypted_field_config` is empty. It may ease readability to abstract this into a named function that does the check i.e. `context_uses_fle2()` rather than relying on comments to explain what the check is doing.

Reply:
Good idea. Added `context_uses_fle2()`